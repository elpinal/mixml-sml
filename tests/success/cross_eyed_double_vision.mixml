[unit
  link C = {module iszero = [val :int -> bool]} with
  link X =
    link t1 = {module t = [type :Type]} with
    link u1 = {module u = [type int]} with
    {module f = [val fun (x : ext t1.t) -> x]}
  with
    link t2 = {module t = [type bool]} with
    link u2 = {module u = [type :Type]} with
    {module g = [val fun (y : ext u2.u) -> ext X.f (ext C.iszero y)]}
]
