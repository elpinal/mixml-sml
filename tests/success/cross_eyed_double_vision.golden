[{f : [bool -> bool]+, g : [int -> bool]+, iszero : [int -> bool]-, t : [= bool], u : [= int]}]+
